asset/asset_test.go

```diff
@@ -12,22 +12,34 @@ import (
 	"github.com/btcsuite/btcd/btcec/v2/schnorr"
 	"github.com/btcsuite/btcd/txscript"
 	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/internal/test"
 	"github.com/lightninglabs/taproot-assets/mssmt"
 	"github.com/lightningnetwork/lnd/keychain"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	hashBytes1     = [32]byte{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
-	hashBytes2     = [32]byte{2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2}
+	hashBytes1     = fn.ToArray[[32]byte](bytes.Repeat([]byte{1}, 32))
+	hashBytes2     = fn.ToArray[[32]byte](bytes.Repeat([]byte{2}, 32))
 	pubKeyBytes, _ = hex.DecodeString(
-		"03a0afeb165f0ec36880b68e0baabd9ad9c62fd1a69aa998bc30e9a346202e078f",
+		"03a0afeb165f0ec36880b68e0baabd9ad9c62fd1a69aa998bc30e9a34620" +
+			"2e078f",
 	)
 	pubKey, _   = btcec.ParsePubKey(pubKeyBytes)
 	sigBytes, _ = hex.DecodeString(
-		"e907831f80848d1069a5371b402410364bdf1c5f8307b0084c55f1ce2dca821525f66a4a85ea8b71e482a74f382d2ce5ebeee8fdb2172f477df4900d310536c0",
+		"e907831f80848d1069a5371b402410364bdf1c5f8307b0084c55f1ce2dca" +
+			"821525f66a4a85ea8b71e482a74f382d2ce5ebeee8fdb2172f47" +
+			"7df4900d310536c0",
 	)
 	sig, _ = schnorr.ParseSignature(sigBytes)
+
+	generatedTestVectorName = "asset_tlv_encoding_generated.json"
+
+	allTestVectorFiles = []string{
+		generatedTestVectorName,
+		"asset_tlv_encoding_error_cases.json",
```

Comment:
Are the `error_cases` tests populated manually, or am I missing where the logic is for generating those?

Reply:
Yeah, so basically if a test vector file name doesn't have a corresponding package-level variable for it (and it only appears in the `allTestVectorFiles` slice) then it means it's currently not autogenerated but contains some manual negative tests.